src/openrct2/ride/station.c

```diff
@@ -281,9 +281,11 @@ static void ride_invalidate_station_start(rct_ride *ride, sint32 stationIndex, s
     if (mapElement == NULL)
         return;
 
-    mapElement->properties.track.sequence &= 0x7F;
-    if (dl != 0)
-        mapElement->properties.track.sequence |= 0x80;
+    mapElement->properties.track.sequence &= ~MAP_ELEM_TRACK_SEQUENCE_GREEN_LIGHT_MASK;
+    if (greenLight)
+    {
+        mapElement->properties.track.sequence |= MAP_ELEM_TRACK_SEQUENCE_GREEN_LIGHT;
```

Comment:
@rwjuk No, it shouldn't. You use a mask to check, not to set.
@IntelOrca I'll change it.